### ISSUE TITLE
[FeG] Cannot find '/etc/magma/configs/service_registry.yml'

### ISSUE THREAD

**Issue by emersonacuna** [magma/magma#5196]
### Your Environment

- **Version:** v1.3.2
- **Affected Component:** FeG
- **Affected Subcomponent:** magmad service
- **Deployment Environment:** local docker-compose (FeG)

### Describe the Issue

**To Reproduce**
After deploy `sudo ./install_gateway feg`, the magmad container is restarting all the time. When we check the logs, we see the following errors:

```
Feb 26 14:34:25 magma-orc8r-test 62907d00fd57[1038]: W0226 14:34:25.681400       1 service_config.go:197] Cannot find '/etc/magma/configs/service_registry.yml'
```
Can see that the directory `/etc/magma/configs` is not creating inside the container. When we check the containers volume we see that it is not mounting:

```
"Mounts": [
            {
                "Type": "bind",
                "Source": "/var/opt/magma/certs",
                "Destination": "/var/opt/magma/certs",
                "Mode": "rw",
                "RW": true,
                "Propagation": "rprivate"
            },
            {
                "Type": "bind",
                "Source": "/var/opt/magma/configs",
                "Destination": "/etc/magma",
                "Mode": "rw",
                "RW": true,
                "Propagation": "rprivate"
            },
            {
                "Type": "bind",
                "Source": "/var/opt/magma/configs/templates",
                "Destination": "/etc/magma/templates",
                "Mode": "rw",
                "RW": true,
                "Propagation": "rprivate"
            },
            {
                "Type": "bind",
                "Source": "/var/opt/magma/configs",
                "Destination": "/var/opt/magma/configs",
                "Mode": "rw",
                "RW": true,
                "Propagation": "rprivate"
            },
            {
                "Type": "bind",
                "Source": "/etc/magma/control_proxy.yml",
                "Destination": "/etc/magma/control_proxy.yml",
                "Mode": "rw",
                "RW": true,
                "Propagation": "rprivate"
            },
            {
                "Type": "bind",
                "Source": "/var/opt/magma/certs/rootCA.pem",
                "Destination": "/var/opt/magma/certs/rootCA.pem",
                "Mode": "rw",
                "RW": true,
                "Propagation": "rprivate"
            }
        ],
```

**Expected behavior**
Containers associated to magmad service must start without any issue

**Additional context**
Consider that we are using the version 1.3.2 of Magma

**Comment by uri200** [comment]
Hey @emersonacuna  can you add here the .env file you used when running `sudo ./install_gateway feg`

Adding slack conversation

https://magmacore.slack.com/archives/G01N7ABCL6M/p1614357723000800?thread_ts=1613955524.002300&cid=G01N7ABCL6M

**Comment by emersonacuna** [comment]
Hi @uri200 here the .env file:
```
# Copyright 2020 The Magma Authors.
# This source code is licensed under the BSD-style license found in the
# LICENSE file in the root directory of this source tree.
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.
COMPOSE_PROJECT_NAME=feg
DOCKER_REGISTRY=docker.io/caprivm/
DOCKER_USERNAME=caprivm
DOCKER_PASSWORD=<token>
IMAGE_VERSION=1.3.2
BUILD_CONTEXT=../../../
ROOTCA_PATH=/var/opt/magma/certs/rootCA.pem
CONTROL_PROXY_PATH=/etc/magma/control_proxy.yml
SNOWFLAKE_PATH=/etc/snowflake
CONFIGS_DEFAULT_VOLUME=/var/opt/magma/configs
CONFIGS_TEMPLATES_PATH=/var/opt/magma/configs/templates
CERTS_VOLUME=/var/opt/magma/certs
CONFIGS_VOLUME=/var/opt/magma/configs
LOG_DRIVER=journald
```

Below you'll find the ```control_proxy.yml``` file:
```
---
#
# Copyright 2020 The Magma Authors.
# This source code is licensed under the BSD-style license found in the
# LICENSE file in the root directory of this source tree.
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.
# nghttpx config will be generated here and used
nghttpx_config_location: /var/tmp/nghttpx.conf
# Location for certs
rootca_cert: /var/opt/magma/certs/rootCA.pem
gateway_cert: /var/opt/magma/certs/gateway.crt
gateway_key: /var/opt/magma/certs/gateway.key
# Listening port of the proxy for local services. The port would be closed
# for the rest of the world.
local_port: 8443
# Cloud address for reaching out to the cloud.
cloud_address: controller.magma.test
cloud_port: 7443
bootstrap_address: bootstrapper-controller.magma.test
bootstrap_port: 7444
fluentd_address: fluentd.magma.test
fluentd_port: 24224
# Option to use nghttpx for proxying. If disabled, the individual
# services would establish the TLS connections themselves.
proxy_cloud_connections: True
# Allows http_proxy usage if the environment variable is present
allow_http_proxy: True
```

We appreciate your help.

**Comment by artblee** [comment]
the mount 
            {
                "Type": "bind",
                "Source": "/var/opt/magma/configs",
                "Destination": "/etc/magma",
                "Mode": "rw",
                "RW": true,
                "Propagation": "rprivate"
            },

might need to be 
            {
                "Type": "bind",
                "Source": "/var/opt/magma/configs",
                "Destination": "/etc/magma/configs",
                "Mode": "rw",
                "RW": true,
                "Propagation": "rprivate"
            },

**Comment by uri200** [comment]
Thanks. As mentioned in slack conversation, I think the location of hte files are proper. My guess is that there is something wrong with the script not copying the files into its appropriate destinations `/etc/magma`. But I can be wrong.

Can you test the changes running the script please?
docs.magmacore.org/docs/feg/deploy_install
Leave the .env file as it is. Docker should build locally the images with that .env file

**Comment by caprivm** [comment]
@uri200 ok, if i understood correctly, the change you want us to make is for us to change the volume that is uploaded to the container just as you suggested @artblee. 

The change that I'm going to be is all the lines that have `${CONFIGS_DEFAULT_VOLUME}:/etc/magma` in `docker-compose.yml`, I'm going to change them for `${CONFIGS_DEFAULT_VOLUME}:/etc/magma/configs`, right?

**Comment by uri200** [comment]
Hi @caprivm  sorry if i was not clear enough. Let me rephrase it.

I don't think the volume or docker-compose is wrong. The reason is that our services on feg do look for configuration files in 
`/etc/magma`
and
`/var/opt/magma/config`

I think the problem is how the `install_gateway.sh feg` command moves the data.

My recommendation is to try the script itself.  

1- Feg is containerized, so you should be able to run in ubuntu/debian. If you have mac, you can try to create a ubuntu vagrant vm or a virtual box with ubuntu/debian. If you have windows I guess you can do the same with virtual box. As long as you have docker inside that virtual box it should work.

2- Before running the script you need a repo with feg docker images already built. You won't have a network repo, lets use your own local repo. To do so just clone magma and do 
```
cd magma/feg/gateway/docker
docker-compose build --parallel
```
This will build feg images and store the images in your local docker repository

3- Now you can try the script following the instructions below
https://docs.magmacore.org/docs/feg/deploy_install
use a different folder than where you cloned magma.
You can use the .env file you had on your clone of magma at `magma/feg/gateway/docker`
You can use any rootCA.pem file (i think there is one in cd .cache/test_certs m but if not just use any)
You can use the control_proxy.yml file provided on the link (or the one you will find in magma)

Once you run the script you will see where it stops (probably building docker?). So that would be the first issue to fix. And so on

**Comment by angelahr450** [comment]
thanks @uri200 , however, thats is exactly the process that we followed, we had to modify the install.sh as you mentioned, the tag v1.3.2 is not compatible with python3.8 into FeG containers... so we made static the tag checkout into the sh (avoiding the scripts comparison), and it worked, However, the script is not creating the /etc/magma/configs.....should we modify the script then?

**Comment by uri200** [comment]
as @joary mentioned in slack chat, please use. 
The linked PR also fixes the documentation


```
ROOTCA_PATH=/var/opt/magma/certs/rootCA.pem
CONTROL_PROXY_PATH=/var/opt/magma/configs/control_proxy.yml
CONFIGS_DEFAULT_VOLUME=/etc/magma
CONFIGS_TEMPLATES_PATH=/etc/magma/templates
SNOWFLAKE_PATH=/etc/snowflake

CERTS_VOLUME=/var/opt/magma/certs
CONFIGS_VOLUME=/var/opt/magma/configs/
```
